cpp/src/arrow/engine/substrait/serde_test.cc

```diff
@@ -5475,7 +5475,7 @@ TEST(Substrait, AsOfJoinDefaultEmit) {
             }
           }
         },
-        "names": ["time", "key", "value1", "time2", "key2", "value2"]
```

Comment:
Why these change?

Reply:
Before the names corresponded to the input indices but now they correspond to the output indices.

Reply:
Actually I see - this is the correct output schema. @rtpsw Did you test this with internal code? I think we also need to update our ibis-substrait producer to produce an emit outputting that matches this.

Reply:
> No, I haven't tested this code internally yet. Do you want this test done before proceeding here? I agree the Substrait producer would have to change.

Yeah I think so. We want to make sure internal code can work with change we make here (with slight modification)

Reply:
This appears to work in internal testing.